front/src/applications/operationalStudies/components/MacroEditor/export.ts

```diff
@@ -0,0 +1,87 @@
+import {
+  osrdEditoastApi,
+  type Distribution,
+  type TrainScheduleBase,
+} from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+import type { NGEDeleteEvent, NGEEvent, NGEUpdateEvent, OperationTrainrunSection } from './types';
+
+// TODO: add a dynamic values when available
+const DEFAULT_PAYLOAD = {
+  constraint_distribution: 'STANDARD' as Distribution,
+  rolling_stock_name: '',
+  start_time: '2024-07-15T08:00:00+02:00',
+};
+
+const createPayloadOperation = (trainrunSections: OperationTrainrunSection[]) => {
+  // TODO: check that the trainrunSections format is still compatible
+  const path = trainrunSections.flatMap((section, index) => {
+    const [sourceTrigram, sourceSecondaryCode] = section.sourceNode.betriebspunktName.split('/');
+    const [targetTrigram, targetSecondaryCode] = section.targetNode.betriebspunktName.split('/');
+    const sourceNode = {
+      trigram: sourceTrigram,
+      secondary_code: sourceSecondaryCode,
+      id: `${section.sourceNode.id}-${index}`,
+    };
+    const targetNode = {
+      trigram: targetTrigram,
+      secondary_code: targetSecondaryCode,
+      id: `${section.targetNode.id}-${index}`,
+    };
```

Comment:
nit: you can create a small function to create a pathItem from a node:
```
const createPathItemFromNode = (
  { betriebspunktName: nodeName, id: nodeId }: Node,
  index: number
) => {
  const [trigram, secondaryCode] = nodeName.split('/');
  return {
    trigram: trigram,
    secondary_code: secondaryCode,
    id: `${nodeId}-${index}`,
  };
};
```
and then 
```suggestion
     const sourceNode = createPathItemFromNode(section.sourceNode, index);
    const targetNode = createPathItemFromNode(section.targetNode, index);
```

Reply:
I had already planned this refacto, but I'm waiting for this [PR](https://github.com/OpenRailAssociation/osrd/pull/8343) to be merged as there will be changes to make on it.